spmd/tensor/api.py

```diff
@@ -219,6 +222,44 @@ def redistribute(
     def local_tensor(self) -> torch.Tensor:
         return self._local_tensor  # type: ignore
 
+    def _view_with_sharding_dim_change(self, sharding_dim, shape):
+        if self.placements[0].is_shard():
+            return self.view(shape)
+        else:
+            if sharding_dim < 0:
+                sharding_dim += self.dim()
+
+            device_mesh = self.device_mesh
+            world_size = device_mesh._dim_groups[0].size()
```

Comment:
nit: can we not using private pg variables to check the world_size? we have a explicit API for us to check the world size for the dim https://github.com/pytorch/PiPPy/blob/1f9ed8cfa93fcbb65d465912d12ad118f35fb86f/spmd/tensor/device_mesh.py#L206

Reply:
oh ok. Sure.